src/models/image-map.ts

```diff
@@ -368,27 +370,24 @@ export const firebaseRealTimeDBImagesHandler: IImageHandler = {
           (url.startsWith(`${kCCImageScheme}://`) && (url.indexOf("concord.org") < 0));
   },
 
-  store(url: string, options?: IImageHandlerStoreOptions) {
-    return new Promise((resolve, reject) => {
-      const { db } = options || {};
-      const { path, classHash, imageKey, normalized } = parseFauxFirebaseRTDBUrl(url);
-
-      if (db && path && normalized) {
-        // In theory we could direct all firebase image requests to the cloud function,
-        // but only cross-class supports require the use of the cloud function.
-        const blobPromise = classHash !== db.stores.user.classHash
-                              ? db.getCloudImageBlob(normalized)
-                              : db.getImageBlob(imageKey);
-        blobPromise.then(blobUrl => {
-          resolve(blobUrl
-                    ? { filename: options?.filename, contentUrl: normalized, displayUrl: blobUrl }
-                    : {});
-        });
-      }
-      else {
-        resolve({});
-      }
-    });
+  async store(url: string, options?: IImageHandlerStoreOptions) {
+    const { db } = options || {};
+    const { path, classHash, imageKey, normalized } = parseFauxFirebaseRTDBUrl(url);
+
+    if (db && path && normalized) {
+      // In theory we could direct all firebase image requests to the cloud function,
+      // but only cross-class supports require the use of the cloud function.
+      const blobPromise = classHash !== db.stores.user.classHash
+                            ? db.getCloudImageBlob(normalized)
+                            : db.getImageBlob(imageKey);
+      const blobUrl = await blobPromise;
```

Comment:
```suggestion
      const blobUrl = classHash !== db.stores.user.classHash
                            ? await db.getCloudImageBlob(normalized)
                            : await db.getImageBlob(imageKey);
```
Doesn't seem like the intermediate step is warranted here.